### PR TITLE
Add building bricks of cnn

### PR DESCRIPTION
**Pull request by hellock**
This PR moves some building bricks from other repos (e.g., mmdetection) to mmcv. Besides, we use registry instead of a dict to manage all modules.
- `ConvModule`
- `build_conv_layer`
- `build_norm_layer`
- `build_activation_layer`
- `build_upsample_layer`
- `build_padding_layer`

### CODE REVIEW

File: mmcv/cnn/bricks/padding.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    'replicate': nn.ReplicationPad2d,
+}
+
+
+def build_padding_layer(cfg, *args, **kwargs):
+    """Build padding layer.
+
+    Args:
+        cfg (None or dict): The padding layer config, which should contain:
+            - type (str): Layer type.
```
Comment (ZwwWayne) [inline mmcv/cnn/bricks/padding.py:15]: We may need to add contents about what we currently support in the docstrings thus the users know what can be used and what needs to be extended.

File: mmcv/cnn/bricks/upsample.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    'pixel_shuffle': PixelShufflePack
+}
+
+
+def build_upsample_layer(cfg):
+    """Build upsample layer.
+
+    Args:
+        cfg (dict): The upsample layer config, which should contain:
+            - type (str): Layer type.
```
Comment (ZwwWayne) [inline mmcv/cnn/bricks/upsample.py:54]: We may need to add avaliable options that we currently support in the docstrings thus the users know what can be used and what needs to be extended.

File: mmcv/cnn/bricks/norm.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    'GN': ('gn', nn.GroupNorm),
+}
+
+
+def build_norm_layer(cfg, num_features, postfix=''):
+    """Build normalization layer.
+
+    Args:
+        cfg (dict): The norm layer config, which should contain:
+            - type (str): Layer type.
```
Comment (ZwwWayne) [inline mmcv/cnn/bricks/norm.py:19]: May also add docstrings about how we can extend them in the downstream repos.

File: mmcv/cnn/bricks/upsample.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+class PixelShufflePack(nn.Module):
+    """ Pixel Shuffle upsample layer
+    Args:
+        in_channels (int): Number of input channels
+        out_channels (int): Number of output channels
+        scale_factor (int): Upsample ratio
+        upsample_kernel (int): Kernel size of Conv layer to expand the channels
+    Returns:
+        upsampled feature map
```
Comment (hejm37) [inline mmcv/cnn/bricks/upsample.py:15]: Docstring format.

File: mmcv/cnn/bricks/scale.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import torch
+import torch.nn as nn
+
+
+class Scale(nn.Module):
+    """
+    A learnable scale parameter
+    """
```
Comment (hellock) [inline mmcv/cnn/bricks/scale.py:8]: Add a standard docstring.
